src/main/java/duke/exceptions/ExceptionChecker.java

```diff
@@ -0,0 +1,179 @@
+        }
+        return inputArray[1];
+    }
+
+    public int retrieveNumberParameter(String[] inputArray) throws DukeException {
+        int taskNumber;
+
+        if (hasNullParameter(inputArray)) {
+            throw new DukeException(MISSING_NUMBER_ERROR);
+        }
+        try {
+            taskNumber = Integer.parseInt(inputArray[1]);
+        } catch (NumberFormatException exception) {
+            throw new DukeException(NUMBER_FORMAT_ERROR);
+        }
+        if (!TaskManager.isValidTaskNumber(taskNumber)) {
+            throw new DukeException(NUMBER_NOT_FOUND_ERROR);
+        }
+
+        return taskNumber;
+    }
+
+    public String retrieveTodoParameter(String[] inputArray) throws DukeException {
+        if (hasNullParameter(inputArray)) {
+            throw new DukeException(TODO_ERROR);
+        }
+        return inputArray[1];
+    }
+
+    public String[] retrieveDeadlineParameters(String[] inputArray) throws DukeException {
+        String[] params;
+
+        if (hasNullParameter(inputArray)) {
+            throw new DukeException(DEADLINE_DESCRIPTION_ERROR);
+        }
+        params = PARSER.separateDeadline(inputArray[1]);
+        if (hasNullParameter(params)) {
+            throw new DukeException(DEADLINE_DATE_ERROR);
+        }
+        if (params[0].isEmpty()) {
+            throw new DukeException(DEADLINE_DESCRIPTION_ERROR);
+        }
+        if (params[1].isEmpty()) {
+            throw new DukeException(DEADLINE_DATE_ERROR);
+        }
+
+        return params;
+    }
+
+    public String[] retrieveEventParameters(String[] inputArray) throws DukeException {
+        String[] params;
+
+        if (hasNullParameter(inputArray)) {
+            throw new DukeException(EVENT_DESCRIPTION_ERROR);
+        }
+        params = PARSER.separateEvent(inputArray[1]);
+        if (hasNullParameter(params)) {
+            throw new DukeException(EVENT_DATE_ERROR);
+        }
+        if (params[0].isEmpty()) {
+            throw new DukeException(EVENT_DESCRIPTION_ERROR);
+        }
+        if (params[1].isEmpty()) {
+            throw new DukeException(EVENT_DATE_ERROR);
+        }
+
+        return params;
+    }
+
+    public void throwInput() throws DukeException {
+        throw new DukeException(INVALID_COMMAND_ERROR);
+    }
+
+    public ArrayList<String> tryToReadFile() throws DukeException {
+        ArrayList<String> fileLines;
+        try {
+            fileLines = FILE_MANAGER.readFile();
+        } catch (FileNotFoundException exception) {
+            try {
+                FILE_MANAGER.createFile();
+            } catch (IOException ioException) {
+                throw new DukeException("IO Exception encountered: " + exception.getMessage());
+            }
+            throw new DukeException(FILE_NOT_FOUND_ERROR);
+        }
+        return fileLines;
+    }
+
+    public ArrayList<String[]> tryToProcessFile(ArrayList<String> fileLines) throws DukeException {
```

Comment:
Again, please use better names.
Creates this question in the mind of the reader: If you are trying to read the file here, where are you reading the file then?